lib/parser/gpx.rb

```diff
@@ -65,19 +65,20 @@ def enhance_trace(points)
     def set_thresholds(points)
       # calculate average speed
       # and assume to slow is 1/10th of it
-      avg_speed = points.sum { |p| p[:speed] } / points.size
+      valid_points = points.select { |p| p[:speed] != Float::INFINITY && p[:speed] != 0 }
+      avg_speed = valid_points.sum { |p| p[:speed] } / valid_points.size
       @to_slow = avg_speed / 10
 
       # get the average durations between the gps points
       # and remove the lower and upper 10% to remove the
       # superfast and the superslow points
       # calculate the average time between the points
       # based on those 80% of the values.
-      durations = points.map { |p| p[:duration] }.sort
-      most_durations = durations[((points.size * 0.1).to_i)..((points.size * 0.9).to_i)]
+      durations = valid_points.map { |p| p[:duration] }.sort
+      most_durations = durations[((valid_points.size * 0.1).to_i)..((valid_points.size * 0.9).to_i)]
 
       # assume the pause has to have a duration > the average * 2
-      @pause_threshold = most_durations.sum / most_durations.size * 2
+      @pause_threshold = most_durations.sum / most_durations.size * 3
```

Comment:
is this even used still? see no reference to it in this class

Reply:
no, that's the point .. yes can remove the calculation also.

Reply:
removed it entirely now